### ISSUE TITLE
ALLELE_SPECIFIC_CALLING:BOOTSTRAPANN restarts on resume

### ISSUE THREAD

**Issue by Jakob37** [genomic-medicine-sweden/tomte#146]
### Description of the bug

When resuming a completed run of Tomte (wanted to run DROP separately after completing all other parts), the pipeline restarts many, but not all of the previously completed `BOOTSTRAPANN` runs.

This means hours of re-processing before reaching DROP.

I am using the dev branch, release 2.1.0, latest commit `00e5eb320ebb6a5a913ff136f45adf7171c7a75a`.

Will go on vacation now, but will be happy to help out trouble-shooting this further when back in two weeks.

**Comment by jemten** [comment]
Thanks for submitting the bug @Jakob37 
Weird that only some of the BOOTSTRAPANN runs are completed 🤔

**Comment by Jakob37** [comment]
> Thanks for submitting the bug @Jakob37 
> Weird that only some of the BOOTSTRAPANN runs are completed 🤔 

All BOOTSTRAPANN runs complete successfully, the issue is when later on resuming downstream parts of the workflow (i.e. to debug DROP), only some of the previously completed are cached, most are rerun. (Maybe this was understood, just wanted to make sure!)

**Comment by jemten** [comment]
Yeah I think some words fell out of my comment 😆

**Comment by Jakob37** [comment]
Doing a fresh run and I still see this issue, with only 11 cached runs being reused out of 98. Will see if I can dig out some more info ...

**Comment by Jakob37** [comment]
Looking in the work dir for the rerun, it seems like the linked in CSV file actually comes from a different sample ID. I.e., it looks the following way (anonymized IDs). 

```
$ ls -l
total 40383
-rw-r--r-- 1 jakob cmd-bnf 41349277 Aug 13 18:07 ID1_ase.vcf
lrwxrwxrwx 1 jakob cmd-bnf      111 Aug 13 18:04 ID1_split_rmdup_info.vcf.gz -> /<work>/f2/7774422ddcf8b5685c4415faf47119/ID1_split_rmdup_info.vcf.gz
lrwxrwxrwx 1 jakob cmd-bnf      115 Aug 13 18:04 ID1_split_rmdup_info.vcf.gz.tbi -> /<work>/f2/7774422ddcf8b5685c4415faf47119/ID1_split_rmdup_info.vcf.gz.tbi
lrwxrwxrwx 1 jakob cmd-bnf       95 Aug 13 18:04 ID2_ase.csv -> /<work>/5d/1c56cb1490752f0ba442fe6aa2371d/ID2_ase.csv
-rw-r--r-- 1 jakob cmd-bnf       93 Aug 13 18:07 versions.yml
```

When looking in the work dir of the original run, the ID matches for the `.csv`.

Will dig some more to see if I can find what might have caused this.

**Comment by Jakob37** [comment]
Hmm. It looks like there might be some mismatch appearing between the channels when I resume the run.

I tried joining `GATK4_ASEREADCOUNTER.out.csv` and `ch_ind_vcf_tbi` prior to running `BOOTSTRAPANN` and it seems like the files are matched correctly now.

I.e. doing this:

```
ch_bootstr_in = ch_ind_vcf_tbi.join(GATK4_ASEREADCOUNTER.out.csv)
BOOTSTRAPANN(ch_bootstr_in)
```

Will complete the process, stop and see if the resume now works correctly (and ls whether the work dirs matches the correct files now).

I don't have a deep enough understanding though of when channels needs to be joined and when they don't, I found this just by testing around.
